### PR TITLE
fix: write retry queue stops retrying

### PR DESCRIPTION
**Pull request by seth-hunter**
Fixes logic error in management of retry buffer that leads to deadlock scenario where write retries cease if server connection is lost for longer than MaxRetryTime

### CODE REVIEW

**Review by vlastahajek — CHANGES_REQUESTED** [review]
Thanks for finding the issue and the fix!

Just a few minor things:
 1. Fix tests, please. You can use this patch [PR332TestFix.zip](https://github.com/influxdata/influxdb-client-go/files/8878499/PR332TestFix.zip) via `git apply`
  2. Change commit messages to be semantic. 
  3. Small change in the error message to be consistent with other messages

File: internal/write/service.go [hunk 1]
```diff
@@ -124,6 +124,16 @@ func (w *Service) HandleWrite(ctx context.Context, batch *Batch) error {
 			log.Debug("Write proc: taking batch from retry queue")
 			if !retrying {
 				b := w.retryQueue.first()
+
+				// Discard batches at beginning of retryQueue that have already expired
+				if time.Now().After(b.Expires) {
+					log.Warn("Write proc: oldest batch in retry queue expired, discarding...")
```
[the remark this answers is not in the document]
Comment (seth-hunter) [reply internal/write/service.go]: These should be resolved in the PR's updated branch.

**Comment by seth-hunter** [comment]
@vlastahajek: thank you for the guidance. Updated branch should address your comments.  In further testing I believe I uncovered an additional related issue and the fix for that has been included in the revised branch.  Requesting your review on that.  Corresponding internal/write tests updated and pass, but I missed an api/write test failure that I will need to digest and think about later (or help appreciated).

**Comment by seth-hunter** [comment]
@vlastahajek: thank you for the review and suggestions. Updated branch should address your comments and also fix an additional closely-related issue I found during further testing. Requesting your review on that, as it may be a little more subjective from a design perspective.

**Review by vlastahajek — APPROVED** [review]

**Comment by vlastahajek** [comment]
Thanks!

**Pull request merged**